sdk/python/kfp/components/placeholders.py

```diff
@@ -15,277 +15,110 @@
 placeholders."""
 
 import abc
-import dataclasses
 import json
-from json.decoder import JSONArray  # type: ignore
-from json.scanner import py_make_scanner
-import re
 from typing import Any, Dict, List, Optional, Union
 
 from kfp.components import base_model
 from kfp.components import utils
 from kfp.components.types import type_utils
 
 
-class Placeholder(abc.ABC):
-    """Abstract base class for Placeholders.
+class Placeholder(abc.ABC, base_model.BaseModel):
```

Comment:
For reviewer:

Subclass `BaseModel` to bypass the `BaseModel`'s logic on which types its attributes can be. This is a workaround and would be good to clean up in another PR.